### ISSUE TITLE
Create/export a command pullImage so 3rd party extensions could call it

### ISSUE THREAD

**Issue by benoitf** [containers/podman-desktop#9989]
### Is your enhancement related to a problem? Please describe

Podman Desktop provides some functionalities and some behavior.

Some extension, in particular bootc extension, want to provides in its UI a button to pull an image but behind this button, it should behave exactly if I would have called the pullImage command from the UI of Podman Desktop.

It means, for example the task should be created and I should see the task in the task manager, etc.

### Describe the solution you'd like

Provides and register the pullImage providing the pull of a given image

### Additional context

While there is an API to pull an image, here the extension want to replicate the behavior of pulling an image from Podman Desktop.

So for example the extension endpoint will not create a task (it's up to the caller of the endpoint), etc.

more discussion around the topic can be found there: https://github.com/podman-desktop/podman-desktop/issues/7398

**Comment by SoniaSandler** [comment]
@benoitf @deboer-tim it looks like even the Podman Desktop UI doesn't have a task/notification for pulling an image. We have tasks for building, saving, or loading images. Am I missing it somewhere?

**Comment by deboer-tim** [comment]
No, we are missing a task - became clear when I opened #10121. Although they are somewhat separate they overlap heavily so I'm going to close #10121 with the expectation that we cover it all here: refactoring the existing UI to use the same task that we expose for extensions.
